docs/20_smart-contracts/30_tutorials/20_create-an-nft.md

```diff
@@ -0,0 +1,769 @@
+      _owners owners(get_self(), token_id);
+
+      // Set the owner of the NFT to the "to" account
+      owners.set(to, get_self());
+
+      // Set the new balance for the "from" account
+      _balances balances(get_self(), from.value);
+      balances.set(balances.get_or_default(0) - 1, get_self());
+
+      // Set the new balance for the "to" account
+      _balances balances2(get_self(), to.value);
+      balances2.set(balances2.get_or_default(0) + 1, get_self());
+
+      // Remove the approval for the "from" account
+      _approvals approvals(get_self(), token_id);
+      approvals.remove();
+
+      // Send the transfer notification
+      require_recipient(from);
+      require_recipient(to);
+   }
+   
+   [[eosio::action]] uint64_t balanceof(name owner){
+      return get_balance(owner);
+   }
+   
+   [[eosio::action]] name ownerof(uint64_t token_id){
+      return get_owner(token_id);
+   }
+   
+   ACTION approve(name to, uint64_t token_id){
+      // get the token owner
+      name owner = get_owner(token_id);
+      
+      // The owner of the NFT must authorize this action
+      check(has_auth(owner), "owner has not authorized the approval");
+   
+      // The account we are approving must exist
+      check(is_account(to), "to account does not exist");
+      
+      // Get the approvals table
+      _approvals approvals(get_self(), token_id);
+      
+      // Set the approval for the NFT
+      approvals.set(to, get_self());
+   }
+   
+   ACTION approveall(name from, name to, bool approved){
+      // The owner of the NFTs must authorize this action
+      check(has_auth(from), "owner has not authorized the approval");
+      
+      // The account we are approving must exist
+      check(is_account(to), "to account does not exist");
+      
+      // Get the approvals table
+      _approvealls approvals(get_self(), from.value);
+      
+      if(approved){
+         // Set the approval for the NFT
+         approvals.set(to, get_self());
+      } else {
+         // Remove the approval for the NFT
+         approvals.remove();
+      }
+   }
+   
+   [[eosio::action]] name getapproved(uint64_t token_id){
+      return get_approved(token_id);
+   }
+   
+   [[eosio::action]] bool approved4all(name owner, name approved_account){
+      return get_approved_all(owner) == approved_account;
+   }
+   
+   [[eosio::action]] std::string gettokenuri(uint64_t token_id){
+      return get_token_uri(token_id);
+   }
+   
+   ACTION setbaseuri(std::string base_uri){
+      // The account calling this action must be the contract owner
+      require_auth(get_self());
+      
+      // Get the base URI table
+      _base_uris base_uris(get_self(), get_self().value);
+      
+      // Set the base URI
+      base_uris.set(base_uri, get_self());
+   }
+};
+```
+</details>
+
+## This is for education purposes
+
+Keep in mind, that if you deployed this contract on the EOS Network and minted tokens, there
```

Comment:
"... and minted tokens" -> " and mint tokens"

Reply:
I think it's correct the way it is, past tense because it's "deployed". Virginia will take a look after 👍